### DISCUSSION TITLE
*ix quandry

### DISCUSSION THREAD

**Discussion by KC5VDJ** [kej715/DtCyber#86] [General]
Okay... The option exists to shift to the unprivileged tcp ports.

I'm getting old and lazy, and people shouldn't be getting a log,4 having to dfterm a fresh (dftermed after build) system just for forgetting sudo ..

Local config option to compile to 4750 perms on target, root owner, open ports asap in main, then suid to user dtcyber (another reason to make suid optional, unprivileged user dtcyber has to be created with a null shell).

This would have to be a local pre-compile OPTION, as I can see situations for both current usage, and well, this option.

I can't stand shoulder surfers...

Or am I missing something?  I'll try it on mine...

It would mean closer source control though, all suid root activity will be centralized to make eyeballing easier, and the whole thing as secure as it can be utilizing suid root in an open source project .  At least the virgin distributions can be trusted in the scenario.

Again, current unprivileged behavior should still be standard, what I want to do should always be presented as an option.

Makefile.*ix:

"# SUID = YES" <-- minus the quotes, # seems to be GitHub formatting

As my starting point.

**Comment by wschaub** [comment]
There's a better way to do this on linux without having to setuid root. 
sudo setcap CAP_NET_BIND_SERVICE=+eip dtcyber will give it the permissions to bind to privileged ports. without granting any other special permissions.  due to how some of the node based services work I do this with the node binary too. 

there's also the  cdcnetPrivilegedTcpPortOffset and dcnetPrivilegedUdpPortOffset cyber.ini options. I think we should support the automation scripts discovering what the offset is and using those ports during install if the user wants to use non-privileged ports.

**Comment by KC5VDJ** [comment]
I was unaware of that one!  Getting caught up is fun.  Anything similar in fbsd these days?  Need to get caught up on that too, been doing fbsd off and on for ages, actually since Dr. Dobbs, actually (Bill Jolitz (sp?)) even emailed me sources.  Always loved the licensing with bsd, and iirc, it should now be wholly unencumbered...with non-bsd licensed code all separate now.

> There's a better way to do this on linux without having to setuid root. 
> sudo setcap CAP_NET_BIND_SERVICE=+eip dtcyber will give it the permissions to bind to privileged ports. without granting any other special permissions.  due to how some of the node based services work I do this with the node binary too. 
> 
> there's also the  cdcnetPrivilegedTcpPortOffset and dcnetPrivilegedUdpPortOffset cyber.ini options. I think we should support the automation scripts discovering what the offset is and using those ports during install if the user wants to use non-privileged ports.
>
